pkg/neg/syncers/utils.go

```
@@ -440,17 +440,17 @@ func validatePod(pod *apiv1.Pod, nodeLister cache.Indexer) bool {
 	// Terminal Pod means a pod is in PodFailed or PodSucceeded phase
 	phase := pod.Status.Phase
 	if phase == apiv1.PodFailed || phase == apiv1.PodSucceeded {
-		klog.V(2).Info("Pod %s/%s is a terminal pod with status %v, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, phase)
+		klog.V(2).Infof("Pod %s/%s is a terminal pod with status %v, skipping", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, phase)
```

these log lines still do not make sense in terms of this utility function. We can leave for a followup PR, but this should probably return an error or just the bool and allow the caller to make a decision about whether to skip or not.

---

Agree. The decision about skipping should be left to the caller. 
I think we already define some errors like `ErrEPNodeNotFound`(endpoint corresponds to an non-existing node), but here we are checking node using Pod. 
If I want to consolidate it and use one error for both endpoints and pods, how should I phrase my error message?